website/project/views/tag.py

```diff
@@ -48,9 +48,37 @@ def project_add_tag(auth, node, **kwargs):
 @must_have_permission('write')
 @must_not_be_registration
 def project_remove_tag(auth, node, **kwargs):
-
     data = request.get_json()
     tag = data['tag']
     if tag:
         node.remove_tag(tag=tag, auth=auth)
         return {'status': 'success'}
+
+
+@must_be_valid_project  # injects project
+@must_have_permission('write')
+@must_not_be_registration
+def file_add_tag(auth, node, guid, **kwargs):
+    data = request.get_json()
+    tag = data['tag']
+    file_name = data['fileName']
+    if tag:
```

Comment:
What happens if `not tag`? That case should return a helpful error.

Pseudocode for a proposed refactor:

```python
data = request.get_json()
tag = data.get('tag')
if not tag:
    return {'status': 'error', 'message': 'Expected key "tag" not provided' }, http.BAD_REQUEST

# <add the tag>
```